site/content/posts/2024-06-05-githubidentityprovider.md

```diff
@@ -0,0 +1,191 @@
+            message: "Only users in certain GitHub teams are allowed to authenticate"
+```
+
+Now users must belong to one of the two teams configured above to be able to successfully authenticate.
+
+## Restricting authentication by GitHub organization membership
+
+Would you rather only allow members of certain GitHub organizations to authenticate? No problem, we've got you covered.
+Just make a small change to your GitHubIdentityProvider.
+
+```yaml
+apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+kind: GitHubIdentityProvider
+metadata:
+  name: my-github-provider
+  namespace: pinniped-supervisor
+spec:
+  allowAuthentication:
+    organizations:
+      policy: OnlyUsersFromAllowedOrganizations
+      allowed:
+        - my-enterprise-organization # this is case-insensitive
+  client:
+    secretName: my-github-provider-client-secret
+```
+
+Now users must belong to the organization configured above to successfully authenticate.
+
+When multiple orgs are `allowed` then the user must belong to any one of those orgs.
+
+Want to further restrict auth by GitHub team membership? No problem, you can still use the technique shown above.
+
+## Using GitHub Enterprise
+
+Are you running GitHub Enterprise for your source control needs? You can use your GitHub Enterprise server's user
+identities by specifying the `host` and optional `tls.certificateAuthorityData`.
+
+```yaml
+apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+kind: GitHubIdentityProvider
+metadata:
+  name: my-github-provider
+  namespace: pinniped-supervisor
+spec:
+  host: github.my-enterprise.example.com
+  tls:
+    certificateAuthorityData: LS0tLS1CRUdJTiBDRVJUSUZJQ0FU.... # optional
+  allowAuthentication:
+    organizations:
+      policy: OnlyUsersFromAllowedOrganizations
+      allowed:
+        - my-enterprise-organization
+  client:
+    secretName: my-github-provider-client-secret
+```
+
+## Kubernetes username and group names
+
+The GitHubIdentityProvider resource offers several choices for how your users' Kubernetes usernames and group names should look.
+
+`.spec.claims.username` allows you to choose from:
+- `login`: The user's GitHub username as shown on their profile. Note that a user can change their own username,
+   so this is not recommended for production use with identities from public github.com.
+- `id`: The numeric user ID assigned by GitHub will be used as the username. On public github.com, this can be found for any user
+   by putting their login name into this GitHub API URL: `https://api.github.com/users/cfryanr` (replace `cfryanr` with the login name).
+   This is automatically assigned and immutable for each user.
+- `login:id`: Blends the readability of using login names with the immutability of using IDs by putting both into
+   the Kubernetes usernames, separated by a colon. This keeps your RBAC policies nicely readable.
+
+`spec.claims.groups` allows you to choose from:
```

Comment:
`.spec` like above?

Reply:
Updated and force pushed. Actually removed the leading `.` from the other one.